src/coreclr/src/jit/gentree.h

```diff
@@ -4303,11 +4439,7 @@ struct GenTreeMultiRegOp : public GenTreeOp
 
     unsigned GetRegCount() const
     {
-        if (GetRegNum() == REG_NA || GetRegNum() == REG_STK)
-        {
-            return 0;
-        }
-        return (gtOtherReg == REG_NA || gtOtherReg == REG_STK) ? 1 : 2;
+        return (TypeGet() == TYP_LONG) ? 2 : 1;
```

Comment:
This method used to return the number of actual registers, rather than the total count, but that is actually less useful. (It may be that this change is where I'm failing on arm32).